integrations/pgvector/src/haystack_integrations/document_stores/pgvector/converters.py

```diff
@@ -51,8 +51,14 @@ def _from_pg_to_haystack_documents(documents: List[Dict[str, Any]]) -> List[Docu
         blob_mime_type = haystack_dict.pop("blob_mime_type")
 
         # convert the embedding to a list of floats
-        if document.get("embedding") is not None:
-            haystack_dict["embedding"] = document["embedding"].tolist()
+        embedding_val = document.get("embedding")
+        converted_embedding = None
+        if embedding_val is not None:
+            if hasattr(embedding_val, "to_list"):  # Check for HalfVector method first
```

Comment:
Could you please explain why we need to call different methods (`to_list` vs `tolist`)?
(I did some research, but I haven't found a satisfactory answer)

Reply:
I have verified this myself. Probably an inconsistent implementation between vector and halfvector in pgvector-psycopg